app/containers/TrackContainer/index.js

```diff
@@ -0,0 +1,258 @@
+import React, { useEffect, memo, useState } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { createStructuredSelector } from 'reselect';
+import { compose } from 'redux';
+import get from 'lodash/get';
+import debounce from 'lodash/debounce';
+import isEmpty from 'lodash/isEmpty';
+import styled from '@emotion/styled';
+import { injectSaga } from 'redux-injectors';
+import { Card, IconButton, Skeleton, InputAdornment, OutlinedInput, CardHeader, Divider } from '@mui/material';
+import { Search as SearchIcon } from '@mui/icons-material';
+import { useHistory } from 'react-router-dom';
+import T from '@components/T';
+import If from '@components/If';
+import For from '@components/For';
+import TrackCard from '@components/TrackCard';
+import colors from '@app/themes/colors';
+import { selectReposData, selectReposError, selectRepoName } from './selectors';
+import { trackContainerCreators } from './reducer';
+import trackContainerSaga from './saga';
+import { translate } from '@app/utils/index';
+
+const CustomCard = styled(Card)`
+  && {
+    margin: 1.25rem 0;
+    padding: 1rem;
+    max-width: ${(props) => props.maxwidth};
+    color: ${(props) => props.color};
+    ${(props) => props.color && `color: ${props.color}`};
+  }
+`;
+const TrackResultsContainer = styled(Card)`
+  && {
+    display: flex;
+    flex-direction: row;
+    flex-wrap: wrap;
+    justify-content: space-around;
+  }
+`;
+const CustomCardHeader = styled(CardHeader)`
+  && {
+    padding: 0;
+  }
+`;
+const Container = styled.div`
+  && {
+    display: flex;
+    flex-direction: column;
+    width: 100%;
+    margin: 0 auto;
+    padding: ${(props) => props.padding}px;
+  }
+`;
+const RightContent = styled.div`
+  display: flex;
+  align-self: flex-end;
+`;
+
+const StyledT = styled(T)`
+  && {
+    color: ${colors.gotoStories};
+  }
+`;
+
+const StyledOutlinedInput = styled(OutlinedInput)`
+  legend {
+    display: none;
+  }
+  > fieldset {
+    top: 0;
+  }
+`;
+
+export function TrackContainer({
+  dispatchItunesTracks,
+  dispatchClearItunesTracks,
+  tracksData,
+  tracksError,
+  trackName,
+  maxwidth,
+  padding
+}) {
+  const [loading, setLoading] = useState(false);
```

Comment:
Instead of adding useState for loading can we shift it to reducer so that it is bounded with your request and success reducer calls and you don't need to manage extra loading in your container